src/features/transfer/useTokenTransfer.ts

```diff
@@ -12,20 +12,159 @@ import { toastTxSuccess } from '../../components/toast/TxSuccessToast';
 import { toWei } from '../../utils/amount';
 import { logger } from '../../utils/logger';
 import { sleep } from '../../utils/timeout';
-import { getErc20Contract } from '../contracts/erc20';
-import { getTokenRouterContract } from '../contracts/hypErc20';
+import { getErc20Contract, getErc721Contract } from '../contracts/token';
+import { getTokenRouterContract } from '../contracts/hypToken';
 import { getMultiProvider, getProvider } from '../multiProvider';
 import { useStore } from '../store';
 import { Route, RouteType, RoutesMap, getTokenRoute } from '../tokens/routes';
 import { isNativeToken } from '../tokens/utils';
 
-import { TransferFormValues, TransferStatus } from './types';
+import { TransferFormValues, ERC721TransferFormValues, TransferStatus } from './types';
 
 // Note, this doesn't use wagmi's prepare + send pattern because we're potentially sending two transactions
 // The prepare hooks are recommended to use pre-click downtime to run async calls, but since the flow
 // may require two serial txs, the prepare hooks aren't useful and complicate hook architecture considerably.
 // See https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/issues/19
 // See https://github.com/wagmi-dev/wagmi/discussions/1564
+export function useERC721Transfer(onDone?: () => void) {
```

Comment:
This has a lot of code duplication with the hook below, it would be good to merge them.